### ISSUE TITLE
Leaderboard

### ISSUE THREAD

**Issue by seifsg** [freeCodeCamp/freeCodeCamp#15899]
Hello,

This is a request for a new feature.

It would be nice to have a leaderboard functionality based on different criteria, by country for example.
@QuincyLarson 

Thanks!

**Comment by dhcodes** [comment]
@seifsg there is actually a [leaderboard project](https://www.freecodecamp.org/challenges/build-a-camper-leaderboard) as part of the DataViz projects that determines ranking by brownie points. With the move the the forum, however, it is less accurate and does apply at the country-level.

https://codepen.io/freeCodeCamp/full/eZGMjp

**Comment by seifsg** [comment]
@dhcodes I meant a tab or some official page to the FCC project itself.
Plus the points system is a bit unfair to those who only want to go for starred map items.
So what I meant is a leaderboard system integrated within the FCC site that counts, for example, the people who finished all certificates or countries with most campers etc..

**Comment by dhcodes** [comment]
@seifsg Thanks for explaining. Would you like to build a prototype?

**Comment by seifsg** [comment]
@dhcodes Yes. But I still don't know so well on how to properly contribute. Are there any restrictions on delivery time?

**Comment by fizzvr** [comment]
Sounds good a internal ranking.

**Comment by raisedadead** [comment]
Hi, @seifsg

However interesting this might sound, we will not have a leaderboard, due to a policy. 
We DO NOT encourage competition via the UX of the platform

That's because most users are new to programming, while it may sound great to have a competing page where you can see who is leading, it's often daunting for newcomers.

Also, any such system can be fooled into motivated rewarding of points. Again a behavior that we would not encourage.

So, to summarize that platforms sole motive is to provide a medium to learn to code for free without distractions of competing. This is very important because of the ideology that we have. 

Thanks once again for your valuable time and interest.

Happy coding!

**Comment by QuincyLarson** [comment]
@seifsg As @raisedadead said, we want to encourage cooperation, rather than competition. This said, you're welcome to create a stand-alone leaderboard system, for the campers who are interested in competing. 

We hope to launch an open API in 2018 that will make this much easier to build and maintain.
